libraries/node-core-library/src/LockFile.ts

```diff
@@ -85,6 +85,28 @@ export class LockFile {
     throw new Error(`File locking not implemented for platform: "${process.platform}"`);
   }
 
+  /**
+   * Attempts to create the lockfile.
+   * Will continue to loop at the given interval until the lock becomes available.
+   * Note that this function is subject to starvation issues.
+   */
+  public static acquire(resourceDir: string, resourceName: string, interval?: number): Promise<LockFile> {
```

Comment:
>interval [](start = 67, length = 8)

The API we want is probably maximumWaitMs.  The polling interval is an implementation detail that nobody but you will know what to do with. :-)